buildconfig/CMake/CommonSetup.cmake

```diff
@@ -241,6 +241,12 @@ if ( ${CMAKE_SYSTEM_NAME} STREQUAL "Linux" )
   include ( LinuxSetup )
 endif ()
 
+###########################################################################
+# Set the c++ standard to 14 - cmake should do the right thing with msvc
+###########################################################################
```

Comment:
CMake 3.10.0 or later with Visual Studio 2015 Update 3 or later do the the right thing.
https://github.com/Kitware/CMake/commit/131c721f542cd6f8805967d847965510172a2c6b

That being said, the earlier behavior isn't "bad" and probably shouldn't be blocked.